src/callbacks/sanitycheck.jl

```diff
@@ -0,0 +1,150 @@
+
+struct Check
+    checkfn
+    name
+    throw_error::Bool
+    message
+end
+
+
+struct SanityCheckException <: Exception end
+
+Base.show(io::IO, check::Check) = print(io, "Check(\"", check.name, "\")")
+
+
+function runchecks(checks, learner)
+    failed = trues(length(checks))
+    for (i, check) in enumerate(checks)
+        failed[i] = !check.checkfn(learner)
+    end
+
+    failedchecks = checks[failed]
+
+    if isempty(failedchecks)
+        return
+    end
+
+    println("$(length(failedchecks))/$(length(checks)) sanity checks failed:")
+
+    for (i, check) in enumerate(failedchecks)
+        println("---")
+        println(i, ": ", check.name, " (", check.error ? "ERROR" : "WARNING", ")")
+        println()
+        println(check.message)
+    end
+
+    if any(getfield.(failedchecks, :throw_error))
+        throw(SanityCheckException())
+    end
+end
+
+
+"""
+    SanityCheck([checks; usedefault = true])
+
+Callback that runs sanity [`Check`](#)s when the `Learner` is initialized.
+If `usedefault` is `true`, it will run all checks in FluxTraining.CHECKS
+in addition to the ones you pass in.
+"""
+struct SanityCheck <: Callback
+    checks::Vector{Check}
+    function SanityCheck(checks = []; usedefault = true)
+        if usedefault
+            checks = vcat(checks, CHECKS)
+        end
+        return new(checks)
+    end
```

Comment:
If you go with the other comment of defining each default check, then I think this makes more sense as defaults for the constructor arguments.

```suggestion
    function SanityCheck(checks = CHECKS; usedefault = false)
        if usedefault
            checks = vcat(checks, CHECKS)
        end
        return new(checks)
    end
```

Reply:
Do you mean `if usedefault and checks != CHECKS`? Alternatively some kind of set union would work too.

Reply:
I was thinking of the following use cases:
- `SanityCheck()`: same as before, use defaults
- `SanityCheck([array of user checks])`: override defaults with user checks
    - this case includes where the user array is a mix and match of defaults
- `SanityCheck([array of user checks]; usedefault = true)`: concat them together

Reply:
This is not that materially different, but I feel like it is more intuitive that when I manually specify an array of checks, it overrides the defaults completely.

Reply:
Of course, my comment was that `SanityCheck(; usedefault = true)` will end up calling `checks = vcat(CHECKS, CHECKS)`.

Reply:
Ah totally missed that. That's a little messy. Maybe this should just be `SanityCheck(checks = []; usedefault = isempty(checks) ? true : false)`.

Reply:
LGTM, think you can even cut that down to `SanityCheck(checks = []; usedefault = isempty(checks))`.

Reply:
Done.